android-components/components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt

```diff
@@ -65,13 +71,22 @@ open class FullScreenFeature(
     override fun onBackPressed(): Boolean {
         val observation = observation
 
-        if (observation.inFullScreen && observation.tabId != null) {
+        if (!isFullscreenCustomTab && observation.inFullScreen && observation.tabId != null) {
```

Comment:
Related to the above comment, if we fold the state holding into the Observation, we just need to check if we're in a custom tab. So we could add a `Observation.isCustomTab` property then.